### PR TITLE
added usage of subtests to the kindergarten-garden exercise

### PR DESCRIPTION
**Pull request by eklatzer**
One of #2098

### CODE REVIEW

File: exercises/practice/kindergarten-garden/kindergarten_garden_test.go [hunk 1]
```diff
@@ -13,120 +14,149 @@ type lookup struct {
...
+	},
+	{
+		number:    4,
+		diagram:   "\nVVCCGG\nVVCCGG",
+		children:  []string{"Alice", "Bob", "Charlie"},
+		errorHint: "",
+		lookups:   []lookup{{"Bob", []string{"clover", "clover", "clover", "clover"}, true}, {"Charlie", []string{"grass", "grass", "grass", "grass"}, true}},
+	},
 	test5, // full garden test
 	test6, // out of order names test
```
[the remark this answers is not in the document]
Comment (eklatzer) [reply exercises/practice/kindergarten-garden/kindergarten_garden_test.go]: The cases are used somewhere else

File: exercises/practice/kindergarten-garden/kindergarten_garden_test.go [hunk 2]
```diff
@@ -13,120 +14,149 @@ type lookup struct {
...
-VVCCGG`, []string{"Alice", "Bob", "Charlie"}, true, []lookup{
-		{"Bob", []string{"clover", "clover", "clover", "clover"}, true},
-		{"Charlie", []string{"grass", "grass", "grass", "grass"}, true},
-	}},
+	{
+		number:    1,
+		diagram:   "\nRC\nGG",
+		children:  []string{"Alice"},
+		errorHint: "",
+		lookups:   []lookup{{"Alice", []string{"radishes", "clover", "grass", "grass"}, true}},
```
[the remark this answers is not in the document]
Comment (eklatzer) [reply exercises/practice/kindergarten-garden/kindergarten_garden_test.go]: I am not sure what you mean, should I add the names of the struct fields or do you mean something else?

[the remark this answers is not in the document]
Comment (andrerfcsantos) [reply exercises/practice/kindergarten-garden/kindergarten_garden_test.go]: Yes, I mean the names of struct fields.

File: exercises/practice/kindergarten-garden/kindergarten_garden_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (andrerfcsantos) [reply exercises/practice/kindergarten-garden/kindergarten_garden_test.go]: Feel free to ignore this suggestion then. I still don't like these tests are being treated specially, but we can deal with this later.

**Review by andrerfcsantos — APPROVED** [review]

**Pull request merged**